ontologytimemachine/custom_proxy.py

```diff
@@ -1,129 +1,117 @@
 from proxy.http.proxy import HttpProxyBasePlugin
-from proxy.http.parser import HttpParser
+from proxy.http.parser import HttpParser, httpParserTypes
 from proxy.common.utils import build_http_response
-import requests
-import sys
+from proxy.http.methods import HttpMethods
+from ontologytimemachine.utils.utils import proxy_logic_http, proxy_logic_https
+from ontologytimemachine.utils.utils import check_if_archivo_ontology_requested
+from ontologytimemachine.utils.utils import get_headers_and_expected_type
+from ontologytimemachine.utils.utils import get_ontology_from_request
+from requests.exceptions import SSLError, Timeout, ConnectionError, RequestException
+from http.client import responses
 import proxy
+import sys
+import requests
+import logging
 
 
 IP = '0.0.0.0'
 PORT = '8899'
 
 
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+
 class OntologyTimeMachinePlugin(HttpProxyBasePlugin):
-    dbpedia_api = 'https://archivo.dbpedia.org/download'
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
 
     def before_upstream_connection(self, request: HttpParser):
-        return None
+        logger.debug('Before upstream')
+        logger.debug(request.method)
+        scheme = 'https' if request.method == b'CONNECT' else 'http'
```

Comment:
it is not https! it is a CONNECT request that has the purpose to tunnel a future https request via a proxy